### PR TITLE
[4.0.6.0] MineDown syntax support; Bug fixes, Optimize finding performance and more

### PR DESCRIPTION
**Pull request by Ghost-chu**
Ready to review

### CODE REVIEW

File: src/main/java/org/maxgamer/quickshop/chat/QuickChatType.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.maxgamer.quickshop.chat.platform.minedown.AdventureQuickChat;
+import org.maxgamer.quickshop.chat.platform.minedown.BungeeQuickChat;
+
+public enum QuickChatType {
+    /*
+     * UNKNOWN = FALLBACK TO REALITEM
+     * REALITEM = USE REAL DROPPED ITEM
+     * ARMORSTAND = USE ARMORSTAND DISPLAY
+     * VIRTUALITEM = USE VIRTUAL DROPPED ITEM (CLIENT SIDE)
+     * */
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/chat/QuickChatType.java:32]: Please update documents

File: src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_RemoveAll.java [hunk 2]
```diff
@@ -56,15 +57,37 @@ public void onCommand(@NotNull CommandSender sender, @NotNull String commandLabe
...
+                }
+                for (Shop shop : tempList) {
+                    if (shop.getOwner().equals(shopOwner.getUniqueId())) {
+                        plugin.log("Deleting shop " + shop + " request by /qs removeall command.");
+                        shop.delete();
+                        i++;
+                    }
+                }
+            } else { //Self shop
+                if (!sender.hasPermission("quickshop.removeall.self")) {
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_RemoveAll.java:75]: Please add this permission change in plugin.yml

File: src/main/java/org/maxgamer/quickshop/util/MsgUtil.java [hunk 3]
```diff
@@ -497,7 +498,7 @@ public static void send(@NotNull UUID player, @NotNull String message, boolean i
             if (p.getPlayer() != null) {
                 if (msgData.length == 3) {
                     try {
-                        sendItemholochat(p.getPlayer(), msgData[0], Objects.requireNonNull(Util.deserialize(msgData[1])), msgData[2]);
+                        plugin.getQuickChat().sendItemHologramChat(p.getPlayer(), msgData[0], Objects.requireNonNull(Util.deserialize(msgData[1])), msgData[2]);
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/util/MsgUtil.java:501]: Caching quickchat instance may better for performance

**Review by sandtechnology — CHANGES_REQUESTED** [review]

File: src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_RemoveAll.java (hunk 2 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_RemoveAll.java:75]: added

File: src/main/java/org/maxgamer/quickshop/util/MsgUtil.java (hunk 3 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/util/MsgUtil.java:501]: QuickChat instance already caching in QuickShop.java

File: src/main/java/org/maxgamer/quickshop/chat/QuickChatType.java (hunk 1 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/chat/QuickChatType.java]: Okay done

**Review by sandtechnology — APPROVED** [review]

**Pull request merged**